src/xcode/ENA/ENA/Source/Scenes/ContactDiary/EditEntries/DiaryEditEntriesViewModel.swift

```diff
@@ -81,25 +83,31 @@ class DiaryEditEntriesViewModel {
 	func removeAll() {
 		switch entryType {
 		case .location:
-			store.removeAllLocations()
+			diaryStore.removeAllLocations()
 		case .contactPerson:
-			store.removeAllContactPersons()
+			diaryStore.removeAllContactPersons()
 		}
 	}
 
 	// MARK: - Private
 
 	private let entryType: DiaryEntryType
-	private let store: DiaryStoringProviding
-
+	private let diaryStore: DiaryStoringProviding
+	private let eventStore: EventStoringProviding
+	
 	private var subscriptions: [AnyCancellable] = []
 
 	private func remove(entry: DiaryEntry) {
 		switch entry {
 		case .location(let location):
-			store.removeLocation(id: location.id)
+			// first remove checkin to this location
+			if let checkinID = location.visit?.checkinId {
+				eventStore.deleteCheckin(id: checkinID)
+			}
```

Comment:
Should there also be a warning for the user that this will delete the checkin, too? Was this specified by any PO?
I could imagine that some users will delete here the location but are not aware that this will delete the checkin, too?